addons/payment_authorize/models/payment.py

```diff
@@ -161,3 +190,74 @@ def _authorize_form_validate(self, data):
                 'acquirer_reference': data.get('x_trans_id'),
             })
             return False
+
+    @api.multi
+    def authorize_s2s_do_transaction(self, **data):
+        self.ensure_one()
+        transaction = AuthhorizeRequest(self.acquirer_id.environment, self.acquirer_id.authorize_login, self.acquirer_id.authorize_transaction_key)
+        tree = transaction.create_authorize_s2s_transaction(self.payment_token_id.acquirer_ref, self.payment_token_id.authorize_payment_id, self.amount, str(self.reference))
+        return self._authorize_s2s_validate_tree(tree)
+
+    @api.multi
+    def _authorize_s2s_validate_tree(self, tree):
+        return self._authorize_s2s_validate(tree)
+
+    @api.multi
+    def _authorize_s2s_validate(self, tree):
+        self.ensure_one()
+        if self.state == 'done':
+            _logger.warning('Authorize: trying to validate an already validated tx (ref %s)' % self.reference)
+            return True
+        status_code = int(tree.get('x_response_code', '0'))
+        if status_code == self._authorize_valid_tx_status:
+            self.write({
+                'state': 'done',
+                'acquirer_reference': tree.get('x_trans_id'),
+            })
+            if self.callback_eval:
+                safe_eval(self.callback_eval, {'self': self})
+            return True
+        elif status_code == self._authorize_pending_tx_status:
+            self.write({
+                'state': 'pending',
+                'acquirer_reference': tree.get('x_trans_id'),
+            })
+            return True
+        elif status_code == self._authorize_cancel_tx_status:
+            self.write({
+                'state': 'cancel',
+                'acquirer_reference': tree.get('x_trans_id'),
+            })
+            return True
+        else:
+            error = tree.get('x_response_reason_text')
+            _logger.info(error)
+            self.write({
+                'state': 'error',
+                'state_message': error,
+                'acquirer_reference': tree.get('x_trans_id'),
+            })
+            return False
+
+
+class PaymentToken(models.Model):
+    _inherit = 'payment.token'
+
+    authorize_payment_id = fields.Char(string='Authorize Payment Reference')
```

Comment:
I found out why: there are 2 references: one for the customer profile, one for the payment profile. It might be interesting to store the customer profile on the partner (or use its id, but risk of collision so everything need to be checked correctly) and the payment profile for the card. This may take some time.